proto/execution_stats.proto

```diff
@@ -135,6 +135,9 @@ message Execution {
 message ExecutionAuxiliaryMetadata {
   // Platform overrides set via remote header.
   build.bazel.remote.execution.v2.Platform platform_overrides = 1;
+
+  // The ExecutionTask that the executor received from the scheduler.
+  build.bazel.remote.execution.v2.ExecutionTask execution_task = 2;
```

Comment:
once we start populating this, we probably want to keep an eye on the prometheus metrics for AC write size and make sure it looks OK

Reply:
Would that be the metric on the right of this screenshot?

![image](https://github.com/user-attachments/assets/5c23920b-69ad-438b-b0a9-7713f9efa2f7)

Reply:
The UI uses the auxiliary metadata (if you were looking for references using `AuxiliaryMetadata` you'll miss the UI references because they are `camelCase`)

Reply:
re. metrics, `Upload throughput` accounts for CAS blobs as well - you could either run a one-off query to restrict to the AC label or update that dashboard to split out those charts by AC vs CAS

Reply:
Ah, that's a good point about the UI. I decided to play it safe and not remove any existing data. Instead, the ExecutionTask will come as a separate auxiliary metadata, which can easily be removed without messing with the other metadata. PTAL.

Now that I don't need the invocation ID, another option is to not use the presence of the ExecutionTask to trigger cache writes from the app. All the necessary data is already available in PublishOperation. One downside of this is we might double write to the cache, especially in the case of self-hosted executors that aren't getting updated.

Reply:
I think the double-write for older executor clients (and during the rollout) seems fine - if we notice it being a problem we could ask people to upgrade and/or gate the logic behind some sort of version check.

Reply:
Here is the version without avoiding double-writes: https://github.com/buildbuddy-io/buildbuddy/pull/8009. Still waiting for Tyler's thoughts on which one he prefers.

Reply:
Tyler's OK with the double writes, so ignore this PR in favor of https://github.com/buildbuddy-io/buildbuddy/pull/8009